File: perl/class/class/src/class/class.pl
#!/usr/bin/perl

# CustomClassをuseで読み込む.
use CustomClass;

# 無名ハッシュリファレンスをクラスと紐付けてオブジェクトにする.
my $obj = {}; # 無名ハッシュ$obj.
bless $obj, CustomClass; # 組み込み関数blessで$objをCustomClassのオブジェクトとする.
$obj->method1(); # $objからCustomClass::method1を呼び出す.
$obj->method2(); # $objからCustomClass::method2を呼び出す.
$obj->method3("ABC"); # #objからCustomClass::method3を呼び出す.(引数に"ABC"を指定.)

File: ruby/class/class/src/class/class.rb
#!/usr/bin/ruby

# クラスTestClassの定義.
class TestClass
  # 指定した値を出力するoutput.
  def output(var)
    puts(var) # putsでvarを出力.
  end
end

# オブジェクトの作成.
obj = TestClass.new # newで作成.
obj.output(10) # obj.outputに10を指定.
